vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -153,7 +141,11 @@ public void setLabel(String label) {
      *            the label html to set
      */
     public void setLabelAsHtml(String htmlContent) {
-        labelElement.getElement().setProperty("innerHTML", htmlContent);
+        // Create and add a new slotted label
```

Comment:
We should call `setLabel("")` here. Otherwise after `setLabel("foo"); setLabelAsHtml("bar")`, `getLabel()` would return "foo". Related https://github.com/vaadin/flow-components/issues/1667

Reply:
Done.